src/main/java/tutorial/Game.java

```diff
@@ -0,0 +1,260 @@
+        ArrayList<Integer> dCards = new ArrayList<>();
+        ArrayList<Integer> cCards = new ArrayList<>();
+        ArrayList<Integer> sCards = new ArrayList<>();
+        Integer[] flushC = new Integer[0];
+
+        for (int i = 0; i < suitArr.length; i++) {
+            if (suitArr[i] == 'H')
+                hCards.add(rankArr[i]);
+            else if (suitArr[i] == 'D')
+                dCards.add(rankArr[i]);
+            else if (suitArr[i] == 'C')
+                cCards.add(rankArr[i]);
+            else
+                sCards.add(rankArr[i]);
+        }
+        if (hCards.size() >= 5) {
+            isFlush = true;
+            flushC = hCards.toArray(new Integer[0]);
+        } else if (dCards.size() >= 5) {
+            isFlush = true;
+            flushC = dCards.toArray(new Integer[0]);
+        } else if (cCards.size() >= 5) {
+            isFlush = true;
+            flushC = dCards.toArray(new Integer[0]);
+        } else if (sCards.size() >= 5) {
+            isFlush = true;
+            flushC = sCards.toArray(new Integer[0]);
+        }
+
+        if (isFlush) {
+            if (checkStraight(flushC)[0] == 1) {
+                return 117 + (flushC[1] == 1 ? flushC[flushC.length - 1] - 1 : 13);
+            }
+        }
+
+        //check pair
+        HashMap<Integer, Integer> nums = new HashMap<Integer, Integer>();
+        for (int i = 0; i < rankArr.length; i++) {
+            if (nums.containsKey(rankArr[i])) {
+                nums.put(rankArr[i], nums.get(rankArr[i]) + 1);
+            } else {
+                nums.put(rankArr[i], 1);
+            }
+        }
+
+        ArrayList<Integer> pairs = new ArrayList<>();
+        ArrayList<Integer> trios = new ArrayList<>();
+
+        for (Integer i : nums.keySet()) {
+            if (nums.get(i) == 2) {
+                is2Same = true;
+                pairs.add(i);
+            } else if (nums.get(i) == 3) {
+                is3Same = true;
+                trios.add(i);
+            } else if (nums.get(i) == 4) {
+                return 91 + (i != 1 ? i - 1 : 13);
+            }
+        }
+
+        Collections.sort(pairs);
+        Collections.sort(trios);
+
+        if (is3Same && is2Same || trios.size() > 1) {
+            return 78 + (trios.get(0) != 1 ? trios.get(trios.size() - 1) - 1 : 13); //this is broken, need to change this
+        } else if (isFlush) {
+            return 65 + (flushC[0] != 1 ? flushC[flushC.length - 1] - 1 : 13);
+        } else if (isStraight) {
+            return 52 + (straightMax != 1 ? straightMax - 1 : 13);
+        } else if (is3Same) {
+            return 39 + (trios.get(0) != 1 ? trios.get(trios.size() - 1) - 1 : 13);
+        } else if (pairs.size() >= 2) {
+            return 26 + (pairs.get(0) != 1 ? pairs.get(pairs.size() - 1) - 1 : 13); //this is also broken
+        } else if (is2Same) {
+            return 13 + (pairs.get(0) != 1 ? pairs.get(pairs.size() - 1) - 1 : 13);
+        }
+

```

Comment:
Code looks good,  should be no problem after this part been fixed. It is great that you have add test while you are writing it.